### PR TITLE
context/config: don't force OpenGL api

### PR DESCRIPTION
**Pull request by kchibisov**
This should allow creating window on android without tweaking the default
for the Api.

--

@MarijnS95 This should solve your issue with `Gles` requests by default.

### CODE REVIEW

File: glutin/src/api/egl/context.rs [hunk 1]
```diff
@@ -33,8 +33,10 @@ impl Display {
         let supports_opengl = self.inner.version > Version::new(1, 3);
 
         let (api, version) = match context_attributes.api {
-            ContextApi::OpenGl(version) if supports_opengl => (egl::OPENGL_API, version),
-            ContextApi::Gles(version) => (egl::OPENGL_ES_API, version),
+            Some(ContextApi::OpenGl(version)) if supports_opengl => (egl::OPENGL_API, version),
+            Some(ContextApi::Gles(version)) => (egl::OPENGL_ES_API, version),
+            None if config.api().contains(Api::OPENGL) => (egl::OPENGL_API, None),
+            None => (egl::OPENGL_ES_BIT, None),
```
Comment (MarijnS95) [inline glutin/src/api/egl/context.rs:39]:
```suggestion
            None => (egl::OPENGL_ES_API
            , None),
```

File: glutin/src/api/egl/config.rs [hunk 2]
```diff
@@ -119,21 +119,23 @@ impl Display {
...
-        }
-        if template.api.contains(Api::GLES3) {
-            api |= egl::OPENGL_ES3_BIT;
-        }
-        if template.api.contains(Api::OPENGL) {
-            api |= egl::OPENGL_BIT;
+        if let Some(requested_api) = template.api {
+            let mut api = 0;
+            if requested_api.contains(Api::GLES1) {
+                api |= egl::OPENGL_ES_API;
```
Comment (MarijnS95) [inline glutin/src/api/egl/config.rs:125]: BIT?

**Comment by MarijnS95** [comment]
Seems okay, besides some BIT/API typos. If I understand correctly, this:

1. Doesn't set any API bit (when not explicitly requested) in the config attributes' `RENDERABLE_TYPE` when creating an EGL display connection, which allows all APIs to be used;
2. Subsequently the EGL display connection can be queried for the exact same attribute, now containing exactly the API(s) it supports (this doesn't seem to be a subset if `RENDERABLE_TYPE` was set explicitly in `ChooseConfig`);
3. Creating a GL context now looks at that list of _actually supported_ APIs to automatically figure out what kind of context to create, defaulting to `OPENGL` if both were supported.

**Comment by kchibisov** [comment]
>creating an EGL display connection, which allows all APIs to be used

When querying the config, display can use multiple apis without any issue.

>Subsequently the EGL display connection can be queried for the exact same attribute, now containing exactly the API(s) it supports (this doesn't seem to be a subset if RENDERABLE_TYPE was set explicitly in ChooseConfig);

You can get the api config supports from it.

>Creating a GL context now looks at that list of actually supported APIs to automatically figure out what kind of context to create, defaulting to OPENGL if both were supported

Yes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Unreleased
 
+- Config doesn't force OpenGL Api by default.
```
Comment (MarijnS95) [inline CHANGELOG.md:3]:
```suggestion
- `Config` doesn't force OpenGL `Api` by default.
```

**Review by MarijnS95 — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
 # Unreleased
 
+- `Config`` doesn't force OpenGL `Api` by default.
```
Comment (MarijnS95) [inline CHANGELOG.md:3]:
```suggestion
- `Config` doesn't force OpenGL `Api` by default.
```

**Pull request merged**